### DISCUSSION TITLE
Run SQL-queries as admin on custom backend

### DISCUSSION THREAD

**Discussion by 98mux** [pocketbase/pocketbase#4633] [Q&A] [answered]
Hi there! Maybe I've missed it but I'd love to execute sql strings from my own backend (seperate from the pocketbase instance).
On my backend I'm logged in as an admin user. Is there any way to run sql queries as a logged in admin user?

Or is extending the pocketbase the only way to accomplish this?

(Also thanks for your hard work, I'm loving Pocketbase so far!)

**Comment by ganigeorgiev** [comment, marked as answer]
There is no support for running raw SQL queries with the Web APIs at the moment (_eventually this may be added as part of https://github.com/pocketbase/pocketbase/issues/2236 but for now remains a very low priority_).

In general it depends on what type of queries you want to execute and how you are planning to use them (if it is a select how the result row(s) will be unmarshalled/scanned?).

If you want to run custom select queries, one possible option could be to create a [View collection](https://pocketbase.io/docs/collections/#view-collection).

Alternatively, if you really want to execute raw SQL queries over a web API, then to some extend you can implement it on your own by registering a custom admin protected route but again the mechanism how the result rows (if any) will be processed is undefined.

**Reply by 98mux** [reply]
Awesome! Thanks for the quick answer! You're awesome :D 
I'll probably implement the admin protected route as you mentioned, so that solves it for me

Optimally it would be awesome to run sql queries something like how the postgres package does it (https://github.com/porsager/postgres): 

```
export const pbAdmin = new PocketBase(PUBLIC_PB_URL);
const authData = await pbAdmin.admins.authWithPassword(PB_ADMIN_EMAIL, PB_ADMIN_PASSWORD);

const age = 18;
const users = await pgAdmin.sql`
    select
      name,
      age
    from users
    where age > ${ age }
  `
```
What i really like about it is that you'd be able to use pocketbase api for all the simple cases (aka 98%) and then just do raw sql for the more advanced stuff
